FileMapper/find_driveid_api.js

```diff
@@ -0,0 +1,234 @@
+        relativePath = path.slice(position + 1) + "\\" + relativePath;
+      }
+      path = path.slice(0, position);
+    }
+  }
+
+  if (!found && relativePath.length > 0) {
+    var currentDirectory = DriveApp.getRootFolder();
+    var relativeMapping = findInDrive(currentDirectory, relativePath);
+
+    if (relativeMapping !== null) {
+      addNewMappingToConfig(localPath, relativeMapping);
+      found = true;
+      driveMapping = relativeMapping;
+    }
+  }
+
+  return driveMapping;
+}
+
+/**
+ * Checks if the mapping for a particular Local path
+ * exists or not in the config
+ * 
+ * @param {String} localPath The local destination path
+ * whose mapping is to be checked
+ * @return {boolean} True if mapping exists, 
+ *                   False otherwise
+ */
+function checkInConfig(localPath) {
+  var documentProperties = PropertiesService.getDocumentProperties();
+  var value = documentProperties.getProperty(localPath);
+  if (value !== null) {
+    return true;
+  }
+  else {
+    return false;
+  }
+}
+
+/**
+ * Get the mapping corresponding to the localPath
+ *
+ * @param {String} localPath The local destination path
+ * whose mapping is to be returned
+ * @return {String} value The corresponding drive Id if mapping exists, 
+ *                  null otherwise
+ */
+function getFromConfig(localPath) {
+  var documentProperties = PropertiesService.getDocumentProperties();
+  var value = documentProperties.getProperty(localPath);
+  if (value !== null) {
+    return value;
+  }
+  else {
+    return null;
+  }
+}
+
+/**
+ * Adds a mapping to the config 
+ * 
+ * @param {String} localPath The local destination 
+ * path
+ * @param {String} id The corresponding drive 
+ * destination Id
+ */
+function addNewMappingToConfig(localPath, id) {
+  var documentProperties = PropertiesService.getDocumentProperties();
+  documentProperties.setProperty(localPath, id);
+}
+
+/**
+ * Find the destination file/folder in the
+ * drive by traversing the given path
+ * 
+ * @param {FolderObject} currentDirectory The Folder 
+ * object for the current directory 
+ * @param {String} path The path to be traversed 
+ * from the current directory to reach the
+ * destination
+ * @return {String} driveId The drive id of the
+ * local file/folder if found or null if file 
+ * is not found
+ */
+function findInDrive(currentDirectory, path) {
+  var folders = path.split("\\");
+
+  if (folders) {
+    var destination = folders[folders.length - 1];
+    var prev = currentDirectory;
+    var curr;
+    var found = false;
+    var driveId = null;
+
+    for (var i = 0; (i < (folders.length - 1) && !found); i++) {
+      curr = folders[i];
+      if (curr && prev) {
+        var fldrs = prev.getFoldersByName(curr);
+        if (fldrs.hasNext()) {
+          prev = fldrs.next();
+        }
+        else {
+          prev = null;
+        }
+      }
+      else {
+        prev = null;
+      }
+      if (prev === null) {
+        found = true;
+      }
+    }
```

Comment:
does making the method recursive simplify it?

Reply:
Didn't make it recursive but i have simplified it. See if it looks good now, otherwise i will try making it recursive.

Reply:
I still think it looks complicated with curr, prev etc. too  many state being maintained

Reply:
Okay, I will make it recursive then.

Reply:
Made it recursive, is it fine now or should i make anymore changes?